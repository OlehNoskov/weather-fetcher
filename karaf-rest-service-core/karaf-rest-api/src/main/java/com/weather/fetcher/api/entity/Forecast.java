package com.weather.fetcher.api.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Forecast {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    private Long id;
    private String country;
    private String city;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd.MM.yyyy hh:mm a")
    private Date date;
    @JsonProperty("weather")
    @OneToOne
    private Weather weather;
}

