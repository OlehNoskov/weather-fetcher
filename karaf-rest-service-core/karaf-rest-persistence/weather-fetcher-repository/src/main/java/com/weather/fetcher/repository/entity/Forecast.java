package com.weather.fetcher.repository.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "forecast")
public class Forecast {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    @Column(name = "id")
    private Long id;
    @Column(name = "country")
    private String country;
    @Column(name = "city")
    private String city;
    @Column(name = "date")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd.MM.yyyy hh:mm a")
    private Date date;
    @Column(name = "weather")
    @JsonProperty("weather")
    @OneToOne
    private Weather weather;
}

