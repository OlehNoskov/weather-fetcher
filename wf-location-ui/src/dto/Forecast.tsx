export interface Forecast {
    country: string,
    city: string,
    date: Date,
    overall: string,
    scale: string,
    degrees: number
}
