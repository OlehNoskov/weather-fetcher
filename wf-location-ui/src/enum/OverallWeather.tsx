export enum OverallWeather {
    SUNNY = 'Sunny',
    CLOUDY = 'Cloudy',
    PARTLY_CLOUDY = 'Partly cloudy',
    RAINY = 'Rainy',
    UNKNOWN = ''
}