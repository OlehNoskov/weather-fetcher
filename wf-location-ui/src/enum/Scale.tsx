export enum Scale {
    CELSIUS = 'CELSIUS',
    FAHRENHEIT = 'FAHRENHEIT'
}
