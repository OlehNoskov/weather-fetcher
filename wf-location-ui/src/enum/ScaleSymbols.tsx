export enum ScaleSymbols {
    CELSIUS = '°C',
    FAHRENHEIT = '℉'
}