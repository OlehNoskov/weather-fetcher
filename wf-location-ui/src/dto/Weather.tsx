import {Temperature} from "./Temperature";
import {OverallWeather} from "../enum/OverallWeather";

export interface Weather {
    overall: OverallWeather,
    temperature: Temperature
}