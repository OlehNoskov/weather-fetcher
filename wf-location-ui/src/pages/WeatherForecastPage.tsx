import React from "react";

import {Card, Typography} from "@mui/material";
import {Forecast} from "../dto/Forecast";
import {OverallWeather} from "../enum/OverallWeather";

import './WeatherForecastPage.css';

interface IForecast {
    forecast: Forecast
}

export default function WeatherForecastPage(props: IForecast) {

    function getWeatherImage(overallWeather: string) {
        switch (overallWeather) {
            case OverallWeather.SUNNY:
                return <img src={"/images/sunny.png"} alt="Sunny"/>;
            case OverallWeather.PARTLY_CLOUDY:
                return <img src={"/images/partly-cloudy.png"} alt="Partly cloudy"/>;
            case OverallWeather.CLOUDY:
                return <img src={"/images/cloudy.png"} alt="Cloudy"/>;
            case OverallWeather.RAINY:
                return <img src={"/images/rainy.png"} alt="Rainy"/>;
            case OverallWeather.UNKNOWN:
                return <img src={"/images/unknown.png"} alt="Unknown weather"/>;
        }
    }

    return (
        <Card className={"weather-card"}>
            <div className={"left-block-info"} data-testid="title">
                <Typography variant="h6" className={"title"}>
                    CURRENT WEATHER
                    <p className={"location"}>{props.forecast.country} {props.forecast.city}</p>
                </Typography>
                <div className={"image-info"}>
                    {getWeatherImage(props.forecast.overall!)}
                </div>
            </div>
            <div className={"degrees"}>
                {props.forecast.degrees} {'Â°C'}
                <p className={"overall-weather"}> {props.forecast.overall}</p>
            </div>
        </Card>
    );
};