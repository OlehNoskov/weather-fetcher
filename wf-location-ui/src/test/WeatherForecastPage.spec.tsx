import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';

import WeatherForecastPage from "../pages/WeatherForecastPage";
import {Forecast} from "../dto/Forecast";
import {OverallWeather} from "../enum/OverallWeather";

const MOCK_FORECAST : Forecast = {
    country: "Poland",
    city: "Warsaw",
    date: new Date(),
    overall: OverallWeather.UNKNOWN,
    scale: 'Celsius',
    degrees: 15
}

describe('LocationPage', () => {

    it('should render WeatherForecastPage without opened modal window unknown weather', () => {
        const {asFragment} = render(<WeatherForecastPage forecast={MOCK_FORECAST}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render WeatherForecastPage without opened modal window sunny weather', () => {
        MOCK_FORECAST.overall = OverallWeather.SUNNY;
        const {asFragment} = render(<WeatherForecastPage forecast={MOCK_FORECAST}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render WeatherForecastPage without opened modal window partly cloudy weather', () => {
        MOCK_FORECAST.overall = OverallWeather.PARTLY_CLOUDY;
        const {asFragment} = render(<WeatherForecastPage forecast={MOCK_FORECAST}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render WeatherForecastPage without opened modal window cloudy weather', () => {
        MOCK_FORECAST.overall = OverallWeather.CLOUDY;
        const {asFragment} = render(<WeatherForecastPage forecast={MOCK_FORECAST}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render WeatherForecastPage without opened modal window rainy weather', () => {
        MOCK_FORECAST.overall = OverallWeather.RAINY;
        const {asFragment} = render(<WeatherForecastPage forecast={MOCK_FORECAST}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render WeatherForecastPage with values', () => {
        MOCK_FORECAST.overall = OverallWeather.RAINY;
        const {asFragment} = render(<WeatherForecastPage forecast={MOCK_FORECAST}/>);

        const location = screen.getByText('Poland Warsaw');
        const degrees = screen.getByText('15 Â°C');
        const overall = screen.getByText('Rainy');

        expect(location).toBeInTheDocument();
        expect(degrees).toBeInTheDocument();
        expect(overall).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
});
