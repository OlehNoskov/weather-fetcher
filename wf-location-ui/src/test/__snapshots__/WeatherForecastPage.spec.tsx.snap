// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationPage should render WeatherForecastPage with values 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root weather-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="left-block-info"
      data-testid="title"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        CURRENT WEATHER
        <p
          class="location"
        >
          Poland Warsaw
        </p>
      </h6>
      <div
        class="image-info"
      >
        <img
          alt="Rainy"
          src="/images/rainy.png"
        />
      </div>
    </div>
    <div
      class="degrees"
    >
      15 °C
      <p
        class="overall-weather"
      >
         Rainy
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocationPage should render WeatherForecastPage without opened modal window cloudy weather 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root weather-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="left-block-info"
      data-testid="title"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        CURRENT WEATHER
        <p
          class="location"
        >
          Poland Warsaw
        </p>
      </h6>
      <div
        class="image-info"
      >
        <img
          alt="Cloudy"
          src="/images/cloudy.png"
        />
      </div>
    </div>
    <div
      class="degrees"
    >
      15 °C
      <p
        class="overall-weather"
      >
         Cloudy
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocationPage should render WeatherForecastPage without opened modal window partly cloudy weather 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root weather-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="left-block-info"
      data-testid="title"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        CURRENT WEATHER
        <p
          class="location"
        >
          Poland Warsaw
        </p>
      </h6>
      <div
        class="image-info"
      >
        <img
          alt="Partly cloudy"
          src="/images/partly-cloudy.png"
        />
      </div>
    </div>
    <div
      class="degrees"
    >
      15 °C
      <p
        class="overall-weather"
      >
         Partly_cloudy
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocationPage should render WeatherForecastPage without opened modal window rainy weather 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root weather-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="left-block-info"
      data-testid="title"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        CURRENT WEATHER
        <p
          class="location"
        >
          Poland Warsaw
        </p>
      </h6>
      <div
        class="image-info"
      >
        <img
          alt="Rainy"
          src="/images/rainy.png"
        />
      </div>
    </div>
    <div
      class="degrees"
    >
      15 °C
      <p
        class="overall-weather"
      >
         Rainy
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocationPage should render WeatherForecastPage without opened modal window sunny weather 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root weather-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="left-block-info"
      data-testid="title"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        CURRENT WEATHER
        <p
          class="location"
        >
          Poland Warsaw
        </p>
      </h6>
      <div
        class="image-info"
      >
        <img
          alt="Sunny"
          src="/images/sunny.png"
        />
      </div>
    </div>
    <div
      class="degrees"
    >
      15 °C
      <p
        class="overall-weather"
      >
         Sunny
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LocationPage should render WeatherForecastPage without opened modal window unknown weather 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root weather-card css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="left-block-info"
      data-testid="title"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
      >
        CURRENT WEATHER
        <p
          class="location"
        >
          Poland Warsaw
        </p>
      </h6>
      <div
        class="image-info"
      >
        <img
          alt="Unknown weather"
          src="/images/unknown.png"
        />
      </div>
    </div>
    <div
      class="degrees"
    >
      15 °C
      <p
        class="overall-weather"
      >
         
      </p>
    </div>
  </div>
</DocumentFragment>
`;
